src/Common/src/Interop/Unix/libgss/Interop.GssApiException.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class libgssapi
+    {
+        internal sealed class GssApiException : Exception
+        {
+            private Status _minorStatus;
+
+            public Status MinorStatus
+            {
+                get { return _minorStatus;  }
+            }
+
+            public GssApiException(string message) : base(message)
+            {
+            }
+
+            public GssApiException(Status majorStatus, Status minorStatus)
+                : this(SR.Format(SR.net_gssapi_operation_failed, majorStatus, minorStatus), majorStatus, minorStatus)
+            {
+            }
+
+            public GssApiException(string message, Status majorStatus, Status minorStatus)
+                : this(message)
+            {
+                HResult = (int)majorStatus;
+                _minorStatus = minorStatus;
+            }
+
+
+            public static GssApiException Create(string message)
+            {
+                return new GssApiException(message);
+            }
+
+            public static GssApiException Create(Status majorStatus, Status minorStatus)
+            {
+                return new GssApiException(majorStatus, minorStatus);
+            }
+
+            public static GssApiException Create(string message, Status majorStatus, Status minorStatus)
+            {
+                return new GssApiException(SR.Format(message, majorStatus, minorStatus), majorStatus, minorStatus);
+            }
+
+            public static void AssertOrThrowIfError(string message, Status majorStatus, Status minorStatus)
+            {
+                if (majorStatus != Status.GSS_S_COMPLETE)
+                {
+                    GssApiException ex = Create(majorStatus, minorStatus);
+                    Debug.Fail(message + ": " + ex);
+                    throw ex;
+                }
+            }
+
+#if DEBUG
+            public override string ToString()
+            {
+                return Message + "\n GSSAPI status: " + GetGssApiDisplayStatus((Status)HResult, _minorStatus);
```

Comment:
@stephentoub In case you missed it, these are in a `#if DEBUG`.  My "yeah!" feeling was extinguished once I realized that.  (Though, it is easy to decide that we wanted that behavior in release, and then forget to localize it)